Smoke tests failures didn't block the run-aqa tests as expected.

We'd like smoke tests failures would block the run-aqa tests.

Now if smoke tests job generation failed the run-aqa tests are blocked.
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk16u/job/jdk16u-windows-x64-openj9/16/console

However if smoke tests job generate correctly but job itself failed, run-aqa test was not blocked as expected.
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk16u/job/jdk16u-linux-x64-hotspot/15/console

Probably need to update https://github.com/adoptium/ci-jenkins-pipelines/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L292

@sophia-guo - we can either:
- disable for dragonwell **or** 
- add alibaba as vendor to Adopt_HS_FeatureTests target at [L35](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/test/functional/buildAndPackage/playlist.xml#L35) if those tests apply **or**
- create a new test target in the playlist for Dragonwell, with the [CommonFeatureTests and/or HotspotFeatureTests](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/test/functional/buildAndPackage/testng.xml#L18-L29)

Also, for the https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-aix-ppc64-hotspot_SmokeTests/1/console failure, [JdkPlatform.javaL65](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/test/functional/buildAndPackage/src/net/adoptopenjdk/test/JdkPlatform.java#L65) needs to be updated to either add or replace the arch string to match on, as os.arch must be returning ppc64 instead of ppc64le

```
13:58:01  Caused by: java.lang.AssertionError: Unrecognized architecture: ppc64
13:58:01  	at net.adoptopenjdk.test.JdkPlatform.detectArchitecture(JdkPlatform.java:90)
13:58:01  	at net.adoptopenjdk.test.JdkPlatform.<init>(JdkPlatform.java:27)
13:58:01  	at net.adoptopenjdk.test.BundledFreetypeTest.<init>(BundledFreetypeTest.java:42)
```

--- a/pipelines/build/common/openjdk_build_pipeline.groovy
+++ b/pipelines/build/common/openjdk_build_pipeline.groovy
@@ -319,22 +319,23 @@ class Build {
                     }
                 }
 
-                context.catchError {
-                    context.build job: jobName,
-                            propagate: true,
-                            parameters: [
-                                    context.string(name: 'SDK_RESOURCE', value: 'upstream'),
-                                    context.string(name: 'UPSTREAM_JOB_NUMBER', value: "${env.BUILD_NUMBER}"),
-                                    context.string(name: 'UPSTREAM_JOB_NAME', value: "${env.JOB_NAME}"),
-                                    context.string(name: 'JDK_VERSION', value: "${jobParams.JDK_VERSIONS}"),
-                                    context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
-                                    context.booleanParam(name: 'KEEP_REPORTDIR', value: buildConfig.KEEP_TEST_REPORTDIR),
-                                    context.string(name: 'ACTIVE_NODE_TIMEOUT', value: "${buildConfig.ACTIVE_NODE_TIMEOUT}"),
-                                    context.booleanParam(name: 'DYNAMIC_COMPILE', value: true),
-                                    context.string(name: 'VENDOR_TEST_BRANCHES', value: vendorTestBranches),
-                                    context.string(name: 'TIME_LIMIT', value: '1')
-                            ]
-                }
+                def testJob = context.build job: jobName,
+                    propagate: false,
+                    parameters: [
+                            context.string(name: 'SDK_RESOURCE', value: 'upstream'),
+                            context.string(name: 'UPSTREAM_JOB_NUMBER', value: "${env.BUILD_NUMBER}"),
+                            context.string(name: 'UPSTREAM_JOB_NAME', value: "${env.JOB_NAME}"),
+                            context.string(name: 'JDK_VERSION', value: "${jobParams.JDK_VERSIONS}"),
+                            context.string(name: 'LABEL_ADDITION', value: additionalTestLabel),
+                            context.booleanParam(name: 'KEEP_REPORTDIR', value: buildConfig.KEEP_TEST_REPORTDIR),
+                            context.string(name: 'ACTIVE_NODE_TIMEOUT', value: "${buildConfig.ACTIVE_NODE_TIMEOUT}"),
+                            context.booleanParam(name: 'DYNAMIC_COMPILE', value: true),
+                            context.string(name: 'VENDOR_TEST_BRANCHES', value: vendorTestBranches),
+                            context.string(name: 'TIME_LIMIT', value: '1')
+                    ]  
+                currentBuild.result = testJob.getResult()
+                return testJob.getResult()
+                     
             }
         } catch (Exception e) {
             context.println "Failed to execute test: ${e.message}"
@@ -463,7 +464,7 @@ class Build {
                             context.node('worker') {
                                 def result = testJob.getResult()
                                 context.echo " ${jobName} result is ${result}"
-                                if (testJob.getResult() == 'SUCCESS' || testJob.getResult() == 'UNSTABLE') {
+                                if (result == 'SUCCESS' || result == 'UNSTABLE') {
                                     context.sh 'rm -f workspace/target/AQAvitTaps/*.tap'
                                     try {
                                         context.timeout(time: 2, unit: 'HOURS') {
@@ -1885,27 +1886,31 @@ class Build {
                 // Run Smoke Tests and AQA Tests
                 if (enableTests) {
                     try {
-                        runSmokeTests()
-                        // Remote trigger Eclispe Temurin JCK tests
-                        if (buildConfig.VARIANT == 'temurin' && isRelease) {
-                            def platform = ''
-                            if (buildConfig.ARCHITECTURE.contains('x64')) {
-                                platform = 'x86-64_' + buildConfig.TARGET_OS
-                            } else {
-                                platform = buildConfig.ARCHITECTURE + '_' + buildConfig.TARGET_OS
-                            }           
-                            if ( !(platform  == 'riscv64_linux' || platform =='aarch64_windows') ) {
-                                if ( !(buildConfig.JAVA_TO_BUILD == 'jdk8u' && platform == 's390x_linux') ) {
-                                    context.echo "Remote trigger Eclipse temurin AQA_Test_Pipeline job with ${platform} ${buildConfig.JAVA_TO_BUILD}"
-                                    def remoteTargets = remoteTriggerJckTests(platform, filename)
-                                    context.parallel remoteTargets
+                        //Only smoke tests succeed TCK and AQA tests will be triggerred.
+                        if (runSmokeTests() == 'SUCCESS') {
+                            // Remote trigger Eclipse Temurin JCK tests
+                            if (buildConfig.VARIANT == 'temurin' && isRelease) {
+                                def platform = ''
+                                if (buildConfig.ARCHITECTURE.contains('x64')) {
+                                    platform = 'x86-64_' + buildConfig.TARGET_OS
+                                } else {
+                                    platform = buildConfig.ARCHITECTURE + '_' + buildConfig.TARGET_OS
+                                }           
+                                if ( !(platform  == 'riscv64_linux' || platform =='aarch64_windows') ) {
+                                    if ( !(buildConfig.JAVA_TO_BUILD == 'jdk8u' && platform == 's390x_linux') ) {
+                                        context.echo "Remote trigger Eclipse Temurin AQA_Test_Pipeline job with ${platform} ${buildConfig.JAVA_TO_BUILD}"
+                                        def remoteTargets = remoteTriggerJckTests(platform, filename)
+                                        context.parallel remoteTargets
+                                    }
                                 }
                             }
-                        }
 
-                        if (buildConfig.TEST_LIST.size() > 0) {
-                            def testStages = runAQATests()
-                            context.parallel testStages
+                            if (buildConfig.TEST_LIST.size() > 0) {
+                                def testStages = runAQATests()
+                                context.parallel testStages
+                            }
+                        } else {
+                            context.println("[ERROR]Smoke tests are not successful! AQA and Tck tests are blocked ")
                         }
                     } catch (Exception e) {
                         context.println(e.message)

